connectionpool.go

```diff
@@ -440,6 +442,11 @@ func (pool *hostConnPool) fill() {
 
 		// mark the end of filling
 		pool.fillingStopped(err != nil)
+
+		if err == nil && startCount == 0 {
+			// notify the session that this node is connected again
+			go pool.session.handleNodeUp(pool.host.ConnectAddress(), pool.port, true)
+		}
```

Comment:
Why is this branch necessary at all? If `startCount==0` then the `go pool.session.handleNodeUp` call was already made on line 433 when we reach this point, wasn't it?

Reply:
Woops, this should've been `startCount > 0` it's meant to be called when we didn't do the synchronous call above. I'll add a comment as well.